### ISSUE TITLE
Direct uploads using ActiveStorage local option don't seem to work

### ISSUE THREAD

**Issue by kstratis** [rails/rails#32187]
tl;dr version:

I'm probably missing something here, but in my setup using ActiveStorage's **local** option with `direct_upload` option set to `true`:

```erb
<div class="form-group">
  <%= property.label :images %>
  <%= property.file_field :images, multiple: true, direct_upload: true, class: 'form-control' %>
</div>
```

... I get the following errors:

```bash
ActiveSupport::MessageVerifier::InvalidSignature (ActiveSupport::MessageVerifier::InvalidSignature):
```
and

```bash
ActiveRecord::RecordNotUnique (PG::UniqueViolation: ERROR:  duplicate key value violates unique constraint "index_active_storage_attachments_uniqueness"
```

I tried doing `rails db:reset` but it was no good.

Is this something to be expected? My goal is to get these nice little progress bars as shown in the [docs](http://edgeguides.rubyonrails.org/active_storage_overview.html#example). Please note that the form works fine if `direct_upload` is set to `false`.

---

### Steps to reproduce

Use ActiveStorage configured to store locally.

```ruby
# development.rb
config.active_storage.service = :local
```

```yml
# storage.yml
local:
  service: Disk
  root: <%= Rails.root.join("storage") %>
  host: "http://shakalaka.lvh.me:3000"

test:
  service: Disk
  root: <%= Rails.root.join("tmp/storage") %>

amazon:
  service: S3
  access_key_id: ""
  secret_access_key: ""
```

```erb
<%#= new.html.erb %>
<div class="form-group">
  <%= property.label :images %>
  <%= property.file_field :images, multiple: true, direct_upload: true, class: 'form-control' %>
</div>
```

### Expected behavior

I'd expect active storage to store the files properly on upload. 

### Actual behavior

```bash
ActiveSupport::MessageVerifier::InvalidSignature (ActiveSupport::MessageVerifier::InvalidSignature):
```
...a couple of lines later

```bash
ActiveRecord::RecordNotUnique (PG::UniqueViolation: ERROR:  duplicate key value violates unique constraint "index_active_storage_attachments_uniqueness"
```

### System configuration
**Rails version**: Rails 5.2.0.rc1
**Ruby version**: Ruby 2.5.0

**Comment by robertlyall** [comment]
Have you [included the JavaScript](http://edgeguides.rubyonrails.org/active_storage_overview.html#direct-upload-installation)?

**Comment by kstratis** [comment]
@robertlyall Yes I've done all the steps. The progress bar thing shows up nicely but never finishes successfully. The request/s return HTTP 500 and the console says what I wrote in the post.

**Comment by georgeclaghorn** [comment]
I’m afraid I couldn’t reproduce in a new application using 5.2.0 RC1. Could you please share a minimal sample application that reproduces the issue you’re seeing?

**Comment by kstratis** [comment]
@georgeclaghorn 
I eventually managed to find the culprit after your suggestion of creating a sample app. 

It turns out that I was mistakenly using this in my controller:

```ruby
def create
    @post = Post.new(post_params)
    @post.images.attach(post_params[:images]) # !!!

    respond_to do |format|
      if @post.save
        format.html { redirect_to @post, notice: 'Post was successfully created.' }
        format.json { render :show, status: :created, location: @post }
      else
        format.html { render :new }
        format.json { render json: @post.errors, status: :unprocessable_entity }
      end
    end
  end
```

By using `Post.new` **and** `attach` I was basically attaching each file twice which explains both the invalid signature (I think) and the duplicate record errors. Maybe the docs could have been a little more helpful on this one but it was mostly and mainly my mistake.

Thank you for your prompt response and my apologies for wasting your time on an issue which was not a Rails bug. Please feel free to close this one.

**Comment by georgeclaghorn** [comment]
No worries. Glad you figured it out.
